upload_vehicle_data.py

```diff
@@ -0,0 +1,729 @@
+        )
+    except subprocess.CalledProcessError as e:
+        logger.error(
+            f'Failed to remove compressed rosbag'
+            f'{remote_compressed_path} from remote machine: {e}'
+        )
+        return False
+
+    return True
+
+
+def get_remote_rosbags_list(logger, remote_user, remote_ip, remote_directory):
+    """Get a list of all rosbags on the remote machine."""
+    list_cmd = f"find {remote_directory} -name '*.mcap'"
+    try:
+        result = subprocess.run(
+            f'ssh {remote_user}@{remote_ip} ' f"'{list_cmd}'",
+            shell=True,
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        rosbag_list = result.stdout.splitlines()
+        logger.info(f'Found {len(rosbag_list)} rosbags on the remote machine.')
+        return rosbag_list
+    except subprocess.CalledProcessError as e:
+        logger.error(f'Failed to list rosbags on remote machine: {e}')
+        raise
+
+
+def list_remote_directories(
+    logger, remote_user, remote_ip, base_remote_directory, depth
+):
+    """List directories on the remote machine up to a given depth."""
+    list_cmd = (
+        f'ssh {remote_user}@{remote_ip} '
+        f'"find {base_remote_directory} -maxdepth {depth} -mindepth 1 -type d"'
+    )
+    try:
+        result = subprocess.run(
+            list_cmd,
+            shell=True,
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        subdirectories = result.stdout.splitlines()
+        logger.info(
+            f'Listed directories up to depth {depth} in '
+            f'{base_remote_directory}'
+        )
+        return subdirectories
+    except subprocess.CalledProcessError as e:
+        logger.error(
+            f'Failed to list subdirectories in {base_remote_directory}: {e}'
+        )
+        return []
+
+
+def get_remote_file_size(logger, remote_user, remote_ip, file_path):
+    """Get the size of a remote file."""
+    size_cmd = f'stat -c%s {file_path}'
+    try:
+        result = subprocess.run(
+            f'ssh {remote_user}@{remote_ip} ' f"'{size_cmd}'",
+            shell=True,
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        file_size = int(result.stdout.strip())
+        logger.info(f'Size of file {file_path} is {file_size} bytes.')
+        return file_size / (1024**3)  # Convert to GB
+    except subprocess.CalledProcessError as e:
+        logger.error(f'Failed to get size of remote file {file_path}: {e}')
+        raise
+
+
+def get_estimated_compression_time(file_sizes):
+    """Estimate compression time in hours based on file sizes."""
+    compression_speed_mbps = 120  # Compression speed in MB/s for zstd level 2
+    total_compression_time = sum(
+        size / compression_speed_mbps / 3600 for size in file_sizes
+    )  # Total compression time in hours
+    return total_compression_time

```

Comment:
I'd be better if you return seconds instead of hours, then you can have better log format as mentioned in other comment

Reply:
Changed